components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -114,11 +134,45 @@ object Glean {
         return "/submit/glean/$docType/$SCHEMA_VERSION/$uuid"
     }
 
-    private fun sendPing(store: String, docType: String) {
+    /**
+     * Collect and assemble the ping. Asynchronously submits the assembled
+     * payload to the designated server using [httpPingUploader].
+     *
+     * @return The [Job] created by the ping submission.
+     */
+    private fun sendPing(store: String, docType: String): Job {
         val pingContent = pingMaker.collect(store)
         val uuid = UUID.randomUUID()
         val path = makePath(docType, uuid)
-        this.httpPingUploader.upload(pingContent, path)
+        // Asynchronously perform the HTTP upload off the main thread.
+        return CoroutineScope(Dispatchers.IO).launch {
+            httpPingUploader.upload(path = path, data = pingContent)
+        }
```

Comment:
Since you asked about Coroutines specifically: This looks good to me. You do not need any fancy cancellation (+ automatic children cancellation) so I think this is good. Instead of creating a new scope on the fly you could also consider using [GlobalScope](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-global-scope/) here. Usually that's something that is discouraged because this launches an unscoped global job that is never cancelled; but that's exactly what you want here.

With GlobalScope that would be something like:

```Kotlin
GlobalScope.launch(Dispatchers.IO) {
    // ...
}
```

If you ever intend to do more work in the background then you could consider creating your own dispatcher. But for now that doesn't seem to be needed.